examples/platforms/emsk/radio.c

```diff
@@ -0,0 +1,668 @@
+
+static RadioPacket sTransmitFrame;
+static RadioPacket sReceiveFrame;
+static RadioPacket sAckFrame;
+
+static ThreadError sTransmitError;
+static ThreadError sReceiveError;
+
+static uint8_t sTransmitPsdu[IEEE802154_MAX_LENGTH];
+static uint8_t sReceivePsdu[IEEE802154_MAX_LENGTH];
+static uint8_t sAckPsdu[IEEE802154_MAX_LENGTH];
+
+static PhyState sState = kStateDisabled;
+static bool sIsReceiverEnabled = false;
+
+static volatile uint8_t Mrf24StatusTx = 0;
+static volatile uint8_t Mrf24StatusRx = 0;
+static volatile uint8_t Mrf24StatusSec = 0;
+
+static DEV_SPI_PTR pmrf_spi_ptr;
+static DEV_GPIO_PTR pmrf_gpio_ptr;
+static void RadioIsr(void *ptr);
+
+/* Variables for test */
+static uint32_t numInterruptRev = 0;
+static uint32_t numInterruptTrans = 0;
+static uint32_t numRadioProcess = 0;
+static uint32_t NODE_ID = 1;
+
+static inline bool isSecurityEnabled(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_SECURITY_ENABLED) != 0;
+}
+
+static inline bool isAckRequested(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_ACK_REQUEST) != 0;
+}
+
+static inline bool isPanIdCompressed(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_PANID_COMPRESSION) != 0;
+}
+
+static inline uint8_t getHeadLength(const uint8_t *frame)
+{
+    uint8_t length = 0;
+    /* Frame Control-2 + Sequence Number-1 */
+    length += 2 + 1;
+
+    /* Destination PAN + Address */
+    switch (frame[1] & IEEE802154_DST_ADDR_MASK)
+    {
+    case IEEE802154_DST_ADDR_SHORT:
+        length += sizeof(otPanId) + sizeof(otShortAddress);
+        break;
+
+    case IEEE802154_DST_ADDR_EXT:
+        length += sizeof(otPanId) + sizeof(otExtAddress);
+        break;
+
+    default:
+        ExitNow(length = 0);
+    }
+
+    /* Source PAN + Address */
+    switch (frame[1] & IEEE802154_SRC_ADDR_MASK)
+    {
+    case IEEE802154_SRC_ADDR_SHORT:
+        if (!isPanIdCompressed(frame))
+        {
+            length += sizeof(otPanId);
+        }
+
+        length += sizeof(otShortAddress);
+        break;
+
+    case IEEE802154_SRC_ADDR_EXT:
+        if (!isPanIdCompressed(frame))
+        {
+            length += sizeof(otPanId);
+        }
+
+        length += sizeof(otExtAddress);
+        break;
+
+    default:
+        ExitNow(length = 0);
+    }
+
+exit:
+    return length;
+}
+
+void enableReceiver(void)
+{
+    if (!sIsReceiverEnabled)
+    {
+        mrf24j40_rxfifo_flush();
+        /* add code to enable receiver (wakeup) */
+        sIsReceiverEnabled = true;
+    }
+}
+
+void disableReceiver(void)
+{
+    if (sIsReceiverEnabled)
+    {
+        mrf24j40_rxfifo_flush();
+        /* add code to disable receiver (sleep) */
+        sIsReceiverEnabled = false;
+    }
+}
+
+void setChannel(uint8_t channel)
+{
+    mrf24j40_set_channel((int16_t)(channel - 11));
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    (void)aInstance;
+    /* should set it manually or preset it in memory */
+    aIeeeEui64[0] = 0x00;
```

Comment:
Yes. It should be a unique number at least. We are trying to read the FPGA chip ID to get a unique number. It requires some additional hardware in the programmable logic. Or a additional memory will be integrated to storage the EUI64 from network chip/device vendors in the future. It is in disscussion. Currently I add a note message in the README.md. The users should add EUI64 in the radio.c. Is it OK?